### PR TITLE
rev to the latest protcol version

### PR DESCRIPTION
**Pull request by devoncarew**
- expose the service protocol timeline API
- add the new `None` event type

@danrubel

### CODE REVIEW

File: dart/example/vm_service_lib_tester.dart [hunk 1]
```diff
@@ -35,8 +35,10 @@ main(List<String> args) async {
   print('dart process started');
 
   process.exitCode.then((code) => print('vm exited: ${code}'));
-  process.stdout.transform(UTF8.decoder).listen(print);
-  process.stderr.transform(UTF8.decoder).listen(print);
+  // TODO: We shouldn't have to use three of these.
+  // ignore: strong_mode_down_cast_composite
```
Comment (devoncarew) [inline dart/example/vm_service_lib_tester.dart]: @pq, I had to add a third ignore here to silence two warnings - it shold just require two (one warning for each of the two lines).

**Comment by danrubel** [comment]
LGTM.

File: dart/example/vm_service_lib_tester.dart (hunk 1 above)
Comment (pq) [inline dart/example/vm_service_lib_tester.dart]: Right.  The gotcha is that ignores work for the following line (EOL comments aren't supported).

File: dart/example/vm_service_lib_tester.dart [hunk 2]
```diff
@@ -35,8 +35,10 @@ main(List<String> args) async {
...
+  process.stdout.transform(UTF8.decoder).listen(print); // ignore: strong_mode_down_cast_composite
```
Comment (pq) [inline dart/example/vm_service_lib_tester.dart]: This ignore is actually impacting the next line.

File: dart/example/vm_service_lib_tester.dart [hunk 3]
```diff
@@ -35,8 +35,10 @@ main(List<String> args) async {
...
+  process.stderr.transform(UTF8.decoder).listen(print); // ignore: strong_mode_down_cast_composite
```
Comment (pq) [inline dart/example/vm_service_lib_tester.dart]: And this one can be removed as it's not doing anything.

File: dart/example/vm_service_lib_tester.dart (hunk 1 above)
Comment (devoncarew) [inline dart/example/vm_service_lib_tester.dart]: Ah, gotcha. I had assumed that it was just operating on the line it was one (just eol comments).

Comment (pq) [inline dart/example/vm_service_lib_tester.dart]: No. @Hixie was confused by this too as I recall.  If you find it too unnatural as is, feel free to open an issue for EOLs.

Comment (Hixie) [inline dart/example/vm_service_lib_tester.dart]: https://github.com/dart-lang/linter/issues/203

**Pull request merged**